File: stage0/build-constants-plugins/src/main/java/net/rubygrapefruit/plugins/stage0/BuildConstantsPlugin.java
package net.rubygrapefruit.plugins.stage0;

import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("unused")
public class BuildConstantsPlugin implements Plugin<Project> {
    @Override
    public void apply(@NotNull Project target) {
        target.getExtensions().create("buildConstants", BuildConstants.class);
    }
}

File: local-plugins/base-plugins/src/main/kotlin/net/rubygrapefruit/plugins/app/internal/plugins/BuildConstantsPlugin.kt
package net.rubygrapefruit.plugins.app.internal.plugins

import net.rubygrapefruit.plugins.app.Versions
import org.gradle.api.Plugin
import org.gradle.api.Project

abstract class BuildConstantsPlugin: Plugin<Project> {
    override fun apply(target: Project) {
        target.run {
            extensions.add("versions", Versions)
        }
    }
}